servers/rendering_server.cpp

```diff
@@ -2870,8 +2870,8 @@ TypedArray<StringName> RenderingServer::_global_shader_parameter_get_list() cons
 }
 
 void RenderingServer::init() {
-	GLOBAL_DEF_RST_NOVAL_BASIC("rendering/textures/vram_compression/import_s3tc_bptc", OS::get_singleton()->get_preferred_texture_format() == OS::PREFERRED_TEXTURE_FORMAT_S3TC_BPTC);
-	GLOBAL_DEF_RST_NOVAL_BASIC("rendering/textures/vram_compression/import_etc2_astc", OS::get_singleton()->get_preferred_texture_format() == OS::PREFERRED_TEXTURE_FORMAT_ETC2_ASTC);
```

Comment:
Shouldn't we change the preferred OS texture format on macOS instead? This will likely make it still look up ETC2/ASTC and thus not work on Intel Mac.

Also this might break the Android editor.

Reply:
Yes, we should also do that. But not instead. To be clear, the project settings themselves must not change between platforms. The behavior can change, sure, but not the booleans themselves, or else you get diffs in `project.godot` when you switch between platforms.

Users of the Android editor will need this setting enabled to avoid diffs compared to desktop platforms, however even with this setting disabled Android will still import what it needs, you'll just get diffs in the .import files. However the difference is that with this setting enabled you can enforce it being imported on all platforms, while in the current master Godot just automatically deletes the setting from project.godot, making it useless.